.github/workflows/pr.yaml

```diff
@@ -0,0 +1,43 @@
+name: pr
+on: [ pull_request ]
+permissions:
+  contents: read
+jobs:
+  semgrep:
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    runs-on: ubuntu-latest
+    container:
+      image: returntocorp/semgrep
+    steps:
+      # Retrieve the source code for the repository
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      # Fetch the semgrep rules
+      - run: git clone https://github.com/dgryski/semgrep-go.git
+      # Run the rule checker using the fetched rules
+      - run: semgrep ci -f semgrep-go
+
+  check-race:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
+        with:
+          # https://www.npmjs.com/package/semver#caret-ranges-123-025-004
+          go-version: '^1.21'
+          check-latest: true
+      - run: go version
+      - run: go test -race ./...
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
+        with:
+          # https://www.npmjs.com/package/semver#caret-ranges-123-025-004
+          go-version: '^1.21'
+          check-latest: true
+      - run: go version
+      - run: go vet ./github ./google ./zalando
+      - run: go test ./...
+      - run: go install honnef.co/go/tools/cmd/staticcheck@latest
+      - run: staticcheck -checks "all"  ./github ./google ./zalando
```

Comment:
Should the staticcheck be in own step. You can use the action from `dominikh/staticcheck-action@xxx`